Intermittent UI test failure - < DownloadFileTest.verifyDownloadedFileOnStorageTest>

### Firebase Test Run: [Firebase link](https://console.firebase.google.com/u/0/project/moz-focus-android/testlab/histories/bh.2189b040bbce6d5a/matrices/8881282099307229585/executions/bs.874305298db176b8/testcases/2)
### Stacktrace:
java.lang.AssertionError: textfile(2).txt finished
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.mozilla.focus.activity.robots.DownloadRobot.verifyDownloadConfirmationMessage(DownloadRobot.kt:51)
	at org.mozilla.focus.activity.DownloadFileTest$verifyDownloadedFileOnStorageTest$3.invoke(DownloadFileTest.kt:205)
	at org.mozilla.focus.activity.DownloadFileTest$verifyDownloadedFileOnStorageTest$3.invoke(DownloadFileTest.kt:202)
	at org.mozilla.focus.activity.robots.DownloadRobotKt.downloadRobot(DownloadRobot.kt:67)
	at org.mozilla.focus.activity.DownloadFileTest.verifyDownloadedFileOnStorageTest(DownloadFileTest.kt:202)
    
### Build:10/19 Main

Flaky 11/3 Main [Firebase link](https://console.firebase.google.com/u/0/project/moz-focus-android/testlab/histories/bh.2189b040bbce6d5a/matrices/7597818670778612449/executions/bs.c1b965d66d3bde50/testcases/2)

java.lang.AssertionError: textfile(2).txt finished
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.mozilla.focus.activity.robots.DownloadRobot.verifyDownloadConfirmationMessage(DownloadRobot.kt:51)
	at org.mozilla.focus.activity.DownloadFileTest$verifyDownloadedFileOnStorageTest$3.invoke(DownloadFileTest.kt:205)
	at org.mozilla.focus.activity.DownloadFileTest$verifyDownloadedFileOnStorageTest$3.invoke(DownloadFileTest.kt:202)
	at org.mozilla.focus.activity.robots.DownloadRobotKt.downloadRobot(DownloadRobot.kt:67)
	at org.mozilla.focus.activity.DownloadFileTest.verifyDownloadedFileOnStorageTest(DownloadFileTest.kt:202)

--- a/app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt
+++ b/app/src/androidTest/java/org/mozilla/focus/helpers/TestHelper.kt
@@ -3,13 +3,10 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 package org.mozilla.focus.helpers
 
-import android.annotation.TargetApi
 import android.app.PendingIntent
 import android.app.UiAutomation
 import android.content.ActivityNotFoundException
 import android.content.ComponentName
-import android.content.ContentResolver
-import android.content.ContentUris
 import android.content.Context
 import android.content.Intent
 import android.content.pm.PackageManager
@@ -19,14 +16,15 @@ import android.graphics.Color
 import android.net.Uri
 import android.os.Build
 import android.os.Build.VERSION.SDK_INT
-import android.os.Bundle
-import android.provider.MediaStore
-import android.provider.MediaStore.setIncludePending
+import android.os.storage.StorageManager
+import android.os.storage.StorageVolume
 import android.util.Log
 import android.view.KeyEvent
+import androidx.annotation.RequiresApi
 import androidx.browser.customtabs.CustomTabColorSchemeParams
 import androidx.browser.customtabs.CustomTabsIntent
 import androidx.browser.customtabs.CustomTabsIntent.SHARE_STATE_ON
+import androidx.core.content.getSystemService
 import androidx.test.espresso.Espresso
 import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.action.ViewActions.click
@@ -49,6 +47,7 @@ import org.mozilla.focus.R
 import org.mozilla.focus.activity.IntentReceiverActivity
 import org.mozilla.focus.ext.getApplicationInfoCompat
 import org.mozilla.focus.utils.IntentUtils
+import java.io.File
 import java.io.FileInputStream
 import java.io.IOException
 import java.io.InputStream
@@ -175,67 +174,14 @@ object TestHelper {
         }
     }
 
+    @RequiresApi(Build.VERSION_CODES.R)
     fun verifyDownloadedFileOnStorage(fileName: String) {
-        val context = getInstrumentation().targetContext
-        val resolver = context.contentResolver
-        val fileUri = queryDownloadMediaStore(fileName)
-
-        val fileExists: Boolean = fileUri?.let {
-            resolver.openInputStream(fileUri).use {
-                it != null
-            }
-        } ?: false
-
-        assertTrue(fileExists)
-    }
-
-    /**
-     * Check the "Downloads" public directory for [fileName] and returns an URI to it.
-     * May be `null` if there is no file with that name in "Downloads".
-     */
-    @TargetApi(Build.VERSION_CODES.Q)
-    private fun queryDownloadMediaStore(fileName: String): Uri? {
-        val context = getInstrumentation().targetContext
-        val resolver = context.contentResolver
-        val queryProjection = arrayOf(MediaStore.Downloads._ID)
-        val querySelection = "${MediaStore.Downloads.DISPLAY_NAME} = ?"
-        val querySelectionArgs = arrayOf(fileName)
-
-        val queryBundle = Bundle().apply {
-            putString(ContentResolver.QUERY_ARG_SQL_SELECTION, querySelection)
-            putStringArray(ContentResolver.QUERY_ARG_SQL_SELECTION_ARGS, querySelectionArgs)
-        }
-
-        // Query if we have a pending download with the same name. This can happen
-        // if a download was interrupted, failed or cancelled before the file was
-        // written to disk. Our logic above will have generated a unique file name
-        // based on existing files on the device, but we might already have a row
-        // for the download in the content resolver.
-        val collection = MediaStore.Downloads.getContentUri(MediaStore.VOLUME_EXTERNAL_PRIMARY)
-        val queryCollection =
-            if (SDK_INT >= Build.VERSION_CODES.R) {
-                queryBundle.putInt(MediaStore.QUERY_ARG_MATCH_PENDING, MediaStore.MATCH_INCLUDE)
-                collection
-            } else {
-                @Suppress("DEPRECATION")
-                setIncludePending(collection)
-            }
-
-        var downloadUri: Uri? = null
-        resolver.query(
-            queryCollection,
-            queryProjection,
-            queryBundle,
-            null,
-        )?.use {
-            if (it.count > 0) {
-                val idColumnIndex = it.getColumnIndex(MediaStore.Downloads._ID)
-                it.moveToFirst()
-                downloadUri = ContentUris.withAppendedId(collection, it.getLong(idColumnIndex))
-            }
-        }
-
-        return downloadUri
+        val storageManager =
+            getInstrumentation().targetContext.getSystemService(Context.STORAGE_SERVICE) as StorageManager
+        val storageVolumes = storageManager.storageVolumes
+        val storageVolume: StorageVolume = storageVolumes[0]
+        val file = File("${storageVolume.directory!!.path}/Download/$fileName")
+        assertTrue(file.exists())
     }
 
     // Method for granting app permission to access location/camera/mic

